### PR TITLE
Prevents Hypo Injecting Synths And Combat Robots.

### PR DESCRIPTION
**Pull request by Wisemonster**
## About The Pull Request
Prevents injecting synths and combat robots with hypos.
## Why It's Good For The Game
Attempting to inject synths/robots currently causes the hypo to "inject" 0 units. This makes it clearer to new players that robots can't use chems, and is overall less confusing.
## Changelog
:cl:
fix: Synths/Robots no longer can be injected with hypos.
/:cl:

### CODE REVIEW

File: code/game/objects/items/reagent_containers/hypospray.dm [hunk 1]
```diff
@@ -70,6 +70,9 @@
 	if(!reagents.total_volume)
 		balloon_alert(user, "Hypospray is Empty.")
 		return
+	if(is_species(A, /datum/species/robot) || is_species(A, /datum/species/synthetic) || is_species(A, /datum/species/early_synthetic))
```
Comment (Lumipharon) [inline code/game/objects/items/reagent_containers/hypospray.dm:73]: Don't make snowflake like this.
These species have traits you can reference instead.

**Review by Lumipharon — CHANGES_REQUESTED** [review]

File: code/game/objects/items/reagent_containers/hypospray.dm (hunk 1 above)
Comment (Wisemonster) [reply code/game/objects/items/reagent_containers/hypospray.dm:73]: Sure, I can use the IS_SYNTHETIC trait for checking synths, though I'm not sure what trait to use for combat robots.

Comment (Lumipharon) [reply code/game/objects/items/reagent_containers/hypospray.dm:73]: robots don't have synthetic afaik.
The more useful trait is the 'don't metabolise chems', whatever its actually called.

Comment (Wisemonster) [reply code/game/objects/items/reagent_containers/hypospray.dm:73]: Don't dead players recieve that trait though?

Comment (Lumipharon) [reply code/game/objects/items/reagent_containers/hypospray.dm:73]: no, they just stop processing

Comment (Wisemonster) [reply code/game/objects/items/reagent_containers/hypospray.dm]: Alright, it should now use flags to check instead of species datums.

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**